### PR TITLE
Add test for preallocated jump stubs for dynamic methods

### PR DESCRIPTION
**Pull request by kouvel**
Prerequisite: dotnet/coreclr#9883

### CODE REVIEW

**Comment by kouvel** [comment]
@jkotas @BruceForstall

File: src/System.Reflection.Emit/tests/DynamicMethodJumpStubTests.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        private static DynamicMethod CreateDynamicMethod(string name)
+        {
+            var dynamicMethod = new DynamicMethod(name, null, null);
+            ILGenerator ilGenerator = dynamicMethod.GetILGenerator();
+            ilGenerator.Emit(OpCodes.Ret);
+            return dynamicMethod;
+        }
+
+        private const uint AllocationGranularity = (uint)64 << 10;
+        private const ulong ReserveRangeRadius = (ulong)4 << 30; // reserve 4 GB before and after the base address
```
Comment (BruceForstall) [inline src/System.Reflection.Emit/tests/DynamicMethodJumpStubTests.cs]: You only need to allocate +/- 2GB

**Comment by kouvel** [comment]
Yea I saw it fail about about half the time when running under Xunit. By itself, the test fails every time.

File: src/System.Reflection.Emit/tests/DynamicMethodJumpStubTests.cs (hunk 1 above)
Comment (kouvel) [inline src/System.Reflection.Emit/tests/DynamicMethodJumpStubTests.cs]: I wanted to allocate more to cover edge cases. Also for reserving memory around framework libraries, I only have one entry point to work with, so the idea was that hopefully the additional 2 GB on each side will cover the rest of the libraries as well.

**Comment by jkotas** [comment]
I think it would be more appropriate for test like this to live in coreclr repo because of it is focused on internal runtime implementation details, not on the API surface

**Comment by kouvel** [comment]
Ok sounds good, I'll add it to the other PR
